pkg/api/folder.go

```diff
@@ -84,8 +86,22 @@ func UpdateFolder(c *models.ReqContext, cmd models.UpdateFolderCommand) response
 	return response.JSON(200, toFolderDto(g, cmd.Result))
 }
 
-func DeleteFolder(c *models.ReqContext) response.Response {
+func (hs *HTTPServer) DeleteFolder(c *models.ReqContext) response.Response { // temporary adding this function to HTTPServer, will be removed from HTTPServer when librarypanels featuretoggle is removed
 	s := dashboards.NewFolderService(c.OrgId, c.SignedInUser)
+	if hs.Cfg.IsPanelLibraryEnabled() {
+		f, err := s.GetFolderByUID(c.Params(":uid"))
```

Comment:
Do you gain anything by first getting the folder with the UID, only to extract the folder ID? Won't it be more performant to query for the folder with the given UID when deleting library panels, instead of querying twice?

Reply:
Hmm, sure I could lookup the id from the dashboards instead. I'll change this. Thank you for suggesting this.

Reply:
Sort of agree, this should be handled in FolderService.DeleteFolder, but that will open a big can of worms. Where PanelLibraryService has a dependency on FolderServce and vice versa. (And FolderService is not a proper DI service) . Can be solved by moving interfaces and models to pkg/models for both FolderService & LibraryPanelService. 

Think this highlights an interesting problem with the design, who owns and maintains the connection between folder and library panel? with folder_id just a column on library_panel it makes sense that library panel owns it. But then we end up with this strange situation where the folder needs know what it's inside it, but that is managed outside it. Hm.. 

Any thoughts?

Reply:
How does folder delete handle dashboards inside the folder? Would be a similar challenge, I think.

Reply:
I am not sure, might be in the folder delete sql handler

Reply:
I moved the FolderID lookup to `deleteLibraryPanelsInFolder` in the LibraryPanels service for now.

Reply:
yes, think that feels best for now, :+1: